src/components/calculator.js

```diff
@@ -1,107 +1,214 @@
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          3
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-brown" type="button">
-          x
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          4
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          5
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          6
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-brown" type="button">
-          -
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          7
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          8
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          9
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-brown" type="button">
-          +
-        </button>
-      </div>
-      <div className="grid-item zero-btn">
-        <button className="calc-btn bg-light-gray" type="button">
-          0
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray text-bold" type="button">
-          .
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-brown" type="button">
-          =
-        </button>
-      </div>
-    </div>
-  </section>
-);
+function Calculator() {
+  const [oldState, newState] = useState({
+    total: 0,
+    next: null,
+    operation: null,
+  });
+  const handleClick = (e) => {
+    newState((oldState) => calculate(oldState, e.target.name));
+  };
+
+  return (
+    <section>
+      <div className="grid-container">
+        <CalcScreen answer={oldState} />
+        <div className="grid-item">
+          <button
+            onClick={handleClick}
+            name="AC"
+            className="calc-btn bg-light-gray"
+            type="button"
+          >
+            AC
+          </button>
+        </div>
+        <div className="grid-item">
+          <button
+            className="calc-btn bg-light-gray"
+            onClick={handleClick}
+            name="-"
+            type="button"
+          >
+            +/-
+          </button>
+        </div>
+        <div className="grid-item">
+          <button
+            className="calc-btn bg-light-gray"
+            onClick={handleClick}
+            name="%"
+            type="button"
```

Comment:
- [ ] Please, kindly ensure that your code is DRY, by not repeating this all along. I think it will be more ideal that your code is dry by trying to put all this repetition in an array, then looping through them, something like

```
const buttonKeys = ['AC', '+/-', '%', '÷', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '=']

<div>
 {buttonKeys.map((buttonkey) => {
                return <button type='button' key={buttonkey.id} onClick={() => handleClick(buttonkey)}>
                      {buttonkey}
                </button>
            })}
</div>
```

See how that long line of code, becomes that short, whoops 😄.
Then the styling can be done using css selectors, you can do that with logic also, but with CSS selectors is less painful 🤣.